src/extension/messages/platformRequest/saveCellExecution.ts

```diff
@@ -91,10 +91,21 @@ export default async function saveCellExecution(
         marshalFrontmatter: true,
       })
 
+      // Fallback to Ephemeral ID if there is no frontmatter
+      if (!notebook?.frontmatter?.runme?.id) {
```

Comment:
Won't using an ephemeral ID undermine the ability to track a document reliably?

I think the solution is to tie the lifecycle ID setting to the relevant features. In other words, document and cell IDs should be written if Stateful features are in play.

Reply:
For Stateful extension, It's not a problem right now because the default settings. 

What you say is to update settings like **Auto-Save** to turn on Lifecycle ID if the Save button is clicked?

Reply:
> What you say is to update settings like Auto-Save to turn on Lifecycle ID if the Save button is clicked?

Yes. Just like all the other former settings, now features, even lifecycle identity is no longer a build-time setting when we blur the lines between the two extension bundles.

Reply:
What will happen if they manually turn of lifecycle but are authed with Stateful?

Reply:
@admc Right now with this PR, it will take the Ephemeral one until a session ends or the file reopens.